src/main.rs

```diff
@@ -21,7 +21,10 @@ async fn main() {
         .expect("Couldn't connect to the database");
 
     let sqlx_repository = SqlxRepository::new(pool);
-    let handler = Handler::new(Arc::new(sqlx_repository));
+    let handler = Handler::new(
+        Arc::new(sqlx_repository.clone()),
+        Arc::new(sqlx_repository.clone()),
+    );
```

Comment:
Daria pra clonar o `Arc` ao invés de clonar o repositorio em si, é mais barato clonar um `Arc` visto que ele é só um ponteiro: 

```suggestion
    let sqlx_repository = Arc::new(SqlxRepository::new(pool));
    
    let handler = Handler::new(
        Arc::clone(sqlx_repository),
        Arc::clone(sqlx_repository),
    );
```

Reply:
rapaz e o sqlx_repository fica como? fica estranho jogar uma ref nele? pq ai dentro do handler eu teria que dar um clone no author e book no método 'new()'